dask/dataframe/shuffle.py

```diff
@@ -156,9 +156,9 @@ def barrier(args):
 def _set_partition(df, index, divisions, p, drop=True):
     """ Shard partition and dump into partd """
     df = df.set_index(index, drop=drop)
-    df = strip_categories(df)
     divisions = list(divisions)
     shards = shard_df_on_index(df, divisions[1:-1])
+    shards = list(map(strip_categories, shards))
```

Comment:
Just a meta-comment since I don't know where else to stick this and it's tangentially related. Adding in some code comments about *why* code is doing what it's doing would be helpful (for me). E.g., `# Barrier` right before the synchronization barrier isn't super helpful. This took me a bit to grok.